charts/cloudflow-provisioning/files/cloudinit/gpu.txt

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+if [[ -f "/etc/node-initialized" ]]; then
+  echo "Node already initialized" >> /var/log/instance_prov.log
+  exit 0
+fi
+sudo swapoff -a
+sudo sed -i '/ swap / s/^/#/' /etc/fstab
+echo "Disabled swap..." >> /var/log/instance_prov.log
+
+sudo ufw disable
+echo "Disabled firewall..." >> /var/log/instance_prov.log
+
+sudo mkdir -p /etc/rancher/rke2
+
+export INSTALL_RKE2_TYPE="agent"
+
+sudo cat <<EOF >>/etc/rancher/rke2/config.yaml
+server: {{ .Values.controlplane.host }}:{{ .Values.controlplane.port }}
+token: {{ .Values.controlplane.token }}
+node-label:
+  - "tenant={{ .Values.tenantId }}"
+  - "cloudflow={{ .Values.cloudflowName }}"
+node-taint:
+  - "tenant={{ .Values.tenantId }}:NoSchedule"
```

Comment:
I need to remind myself that we'll need to modify all the generated JSONS for deployment to add this for the vcluster based CFs.

Reply:
So vcluster has a nice little parameter...
```
extraArgs:
    - --enforce-toleration=tenant=test:NoSchedule
```
This will, for any workload that flows through a vcluster controlplane, automatically tag on the tolerations listed without them needing to be listed in the YAML